Dispatcher/UI/Homepage/HomepageViewController.swift

```diff
@@ -126,8 +96,10 @@ extension HomepageViewController: UIScrollViewDelegate {
         
         let position = scrollView.contentOffset.y
         if position > (tableView.contentSize.height - 100 - scrollView.frame.size.height) {
-            fetchNewsFromAPI() {
+            viewModel.fetchNewsFromAPI() {
+                self.dataSource.models = self.viewModel.newsArray
```

Comment:
I Don't think you should set the data source again since you already looking on this array as data source in line 29.
Just reload the data.

Reply:
It doesn't work without it
@abenartz